infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/primitive/SszUInt64.java

```diff
@@ -19,9 +19,23 @@
 
 public class SszUInt64 extends AbstractSszPrimitive<UInt64, SszUInt64> {
 
-  public static final SszUInt64 ZERO = SszUInt64.of(UInt64.ZERO);
+  public static final SszUInt64 ZERO = SszUInt64.valueOf(UInt64.ZERO);
+  public static final SszUInt64 THIRTY_TWO_GWEI = SszUInt64.valueOf(UInt64.THIRTY_TWO_GWEI);
+
+  public static final SszUInt64 MAX_VALUE = SszUInt64.valueOf(UInt64.MAX_VALUE);
 
   public static SszUInt64 of(UInt64 val) {
+    if (val.isZero()) {
+      return ZERO;
+    } else if (val.isThirtyTwoGwei()) {
```

Comment:
you don't need `else` here

Reply:
true enough i could just do if, return etc